lib/storage/bool_preference_storage.dart

```diff
@@ -9,17 +9,17 @@ abstract class BoolPreference {
     await _instance.setBool(key, !current);
   }
 
-  Future<void> setBool({bool value}) async {
+  Future<void> setBool({bool? value}) async {
```

Comment:
we don't expect to store null in share preference. If it is a null value, we remove it directly using method `clear()`

result should be done by remove that ? from bool, `setBool({required bool value}){}`